src/DeviceInterfaceValues.tsx

```diff
@@ -184,6 +189,346 @@ const InterfaceData = ({ interfaceData }: InterfaceDataProps): React.ReactElemen
+          )
+          .required('This field is required and must be an array of datetime strings.')
+          .typeError('Each item in the array must be a valid datetime string.');
+      default:
+        return yup.string().required('This field is required.');
+    }
+  };
+
+  const parseValue = (type: string, value: string) => {
+    switch (type) {
+      case 'integer':
+      case 'longinteger':
+        return parseInt(value, 10);
+      case 'double':
+        return parseFloat(value);
+      case 'boolean':
+        return value.toLowerCase() === 'true';
+      case 'doublearray':
+      case 'integerarray':
+      case 'booleanarray':
+      case 'longintegerarray':
+      case 'stringarray':
+      case 'binaryblobarray':
+      case 'datetimearray':
+        return JSON.parse(value);
+      default:
+        return value;
+    }
+  };
+
+  const handleSelectedMapping = (e: ChangeEvent<HTMLSelectElement>) => {
+    const selected = interfaceData?.mappings.find((mapping) => mapping.endpoint === e.target.value);
+    setSelectedMapping((selected as AstarteMapping) || null);
+    setEndpoint(e.target.value);
+    setValue('');
+    setParamValues({});
+    setErrors({});
+  };
+
+  const handleValueChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+    setErrors({});
+  };
+
+  const handleParamChange = (paramName: string, paramValue: string) => {
+    setParamValues((prevValues) => ({ ...prevValues, [paramName]: paramValue }));
+  };
+
+  const handleObjectData = (dataName: string, dataValue: string) => {
+    const mapping = interfaceData?.mappings.find((m) => {
+      const segments = m.endpoint.split('/');
+      return segments[segments.length - 1] === dataName;
+    });
+    if (mapping) {
+      setData((prevValues) => ({
+        ...prevValues,
+        [dataName]: dataValue,
+      }));
+      const schema = getValidationSchema(mapping.type);
+      schema
+        .validate(dataValue)
+        .then(() => {
+          setParsedObjectData((prevValues) => ({
+            ...prevValues,
+            [dataName]: parseValue(mapping.type, dataValue),
+          }));
+          setErrors((prevErrors) => ({ ...prevErrors, [dataName]: '' }));
+        })
+        .catch((err) => {
+          setErrors((prevErrors) => ({ ...prevErrors, [dataName]: err.message }));
+        });
+    }
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    sendInterfaceData({
+      endpoint: endpointWithParams,
+      value: interfaceData?.aggregation === 'object' ? parsedObjectData : parsedIndividualValue,
+    });
+  };
+
+  useEffect(() => {
+    const pathParamsArray: string[] = [];
+    const endpointParams = endpoint.match(/%\{(.+?)\}/g);
+    if (endpointParams) {
+      endpointParams.forEach((param) => {
+        const paramName = param.slice(2, -1);
+        pathParamsArray.push(paramName);
+      });
+    }
+    setPathParams(pathParamsArray);
+  }, [endpoint]);
```

Comment:
how to extract parameters from a mapping should be something that we write in the mapping model.
We could use an utility function like:
```typescript
function getEndpointParameters(endpoint: string) { ... }
```